sources/engine/Stride.Physics/Simulation.cs

```diff
@@ -1066,12 +1053,67 @@ internal void Simulate(float deltaTime)
             OnSimulationEnd(simulationArgs);
         }
 
+        public class SimulationArgs : EventArgs
+        {
+            public float DeltaTime;
+        }
+
+        /// <summary>
+        /// Called before the physics simulation.
+        /// This event might not be fired by the main thread.
+        /// </summary>
+        [Obsolete($"The simulation is not guaranteed to tick following this call, use {nameof(PreTick)} instead. This is the same thing as Update")]
+        public event EventHandler<SimulationArgs> SimulationBegin;
+
         /// <summary>
-        /// Called right after the physics simulation.
+        /// Called after the physics simulation.
         /// This event might not be fired by the main thread.
         /// </summary>
+        [Obsolete($"The simulation is not guaranteed to have ticked before this call, use {nameof(PostTick)} instead. This is the same thing as Update")]
         public event EventHandler<SimulationArgs> SimulationEnd;
 
+        /// <summary>
+        /// Called right before processing a tick of the physics simulation,
+        /// this may occur from zero up to <see cref="MaxSubSteps"/> time per update.
+        /// </summary>
+        public event SimulationTickEvent PreTick
+        {
+            add
+            {
+                if (preSimulationTick is null)
+                    discreteDynamicsWorld.SetInternalTickCallback((world, step) => preSimulationTick?.Invoke(this, step), new object(), isPreTick:true);
```

Comment:
For this `new object()` - is that required to be an instance?
Could `null` or some static object instance be used instead?

Reply:
Thankfully this is just a single allocation for the lifetime of `Simulation`, but I'll take a look tomorrow anyway